marconi-chain-index/src/Marconi/ChainIndex/Experimental/Run.hs

```diff
@@ -25,25 +33,48 @@ run appName = do
       includeScript = True
       socketPath = Cli.optionsSocketPath $ Cli.commonOptions o
       networkId = Cli.optionsNetworkId $ Cli.commonOptions o
+      retryConfig = Cli.optionsRetryConfig $ Cli.commonOptions o
+      preferredStartingPoint = Cli.optionsChainPoint $ Cli.commonOptions o
   createDirectoryIfMissing True (Cli.optionsDbPath o)
   trace <- defaultStdOutLogger appName
+
+  securityParam <- withNodeConnectRetry trace retryConfig socketPath $ do
+    Utils.toException $ Utils.querySecurityParam @Void networkId socketPath
+
   mindexers <-
     runExceptT $
       buildIndexers
+        securityParam
         (Core.CatchupConfig batchSize stopCatchupDistance)
         (Utxo.UtxoIndexerConfig filteredAddresses includeScript)
         (MintTokenEvent.MintTokenEventConfig filteredAssetIds)
         trace
         (Cli.optionsDbPath o)
-  let (_utxoQueryIndexer, indexers) = case mindexers of
-        Left err -> error $ show err
-        Right result -> result
-  let retryConfig = Cli.optionsRetryConfig $ Cli.commonOptions o
+  (indexerLastSyncPoints, _utxoQueryIndexer, indexers) <-
+    ( case mindexers of
+        Left err -> do
+          logError trace $ Text.pack $ show err
+          exitFailure
+        Right result -> pure result
+      )
+
+  let startingPoints = getStartingPoints preferredStartingPoint indexerLastSyncPoints
+
   logInfo trace $ appName <> "-" <> Text.pack Cli.getVersion
+
   Runner.runIndexer
     trace
+    securityParam
     retryConfig
     socketPath
     networkId
-    (Cli.optionsChainPoint $ Cli.commonOptions o)
+    startingPoints
     indexers
+
+getStartingPoints :: C.ChainPoint -> [C.ChainPoint] -> NonEmpty C.ChainPoint
+getStartingPoints preferredStartingPoint indexerLastSyncPoints =
+  case preferredStartingPoint of
+    C.ChainPointAtGenesis ->
+      fromMaybe (NonEmpty.singleton C.ChainPointAtGenesis) $
+        NonEmpty.nonEmpty indexerLastSyncPoints
```

Comment:
Can't we use `C.ChainPointAtGenesis` if it's provided (I don't think we have a CLI option for it but we could add one in the future).

Reply:
True. In the current CLI, if we don't provide a `ChainPoint` as a CLI option, we assume its genesis. So yeah, that should be reworked in the future story about improving this piece of the CLI.